pyaedt/modules/solutions.py

```diff
@@ -1438,743 +1357,647 @@ def assign_weight(self, a, b, taper="flat", port_cont=0):
-        # PhaseShiftA and PhaseShiftB, given Wave Vector (k), lattice vectors
-        # (Ax, Ay, Bx, By), Scan angles (theta, phi) using formula below
-        # Phase Shift A = - (Ax*k*sin(theta)*cos(phi) + Ay*k*sin(theta)*sin(phi))
-        # Phase Shift B = - (Bx*k*sin(theta)*cos(phi) + By*k*sin(theta)*sin(phi))
-        # ------------------------------------------------------------------------
+    # fmt: off
+    @pyaedt_function_handler()
+    def plot_2d_cut(
+            self,
+            farfield_quantity="RealizedGain",
+            primary_sweep="phi",
+            secondary_sweep_value=0,
+            phi_scan=0,
+            theta_scan=0,
+            title="Far Field Cut",
+            format_quantity="dB10",
+            export_image_path=None,
+            show=True,
+            is_polar=False,
+            **kwargs
+    ):
+        # fmt: on
+        """Create a 2D plot of a specified quantity in Matplotlib.
 
-        theta_scan1 = math.radians(theta_scan1)
-        phi_scan1 = math.radians(phi_scan1)
+        Parameters
+        ----------
+        farfield_quantity : str, optional
+            Quantity to plot. The default is ``"RealizedGain"``.
+            Available functions are: ``"RealizedGain"``, ``"RealizedGain_Theta"``, ``"RealizedGain_Phi"``,
+            ``"rETotal"``, ``"rETheta"``, and ``"rEPhi"``.
+        primary_sweep : str, optional.
+            X axis variable. The default is ``"phi"``. Options are ``"phi"`` and ``"theta"``.
+        secondary_sweep_value : float, list, string, optional
+            List of cuts on the secondary sweep to plot. The default is ``0``. Options are
+            `"all"`, a single value float, or a list of float values.
+        phi_scan : float, int, optional
+            Phi scan angle in degrees. The default is ``0``.
+        theta_scan : float, int, optional
+            Theta scan angle in degrees. The default is ``0``.
+        title : str, optional
+            Plot title. The default is ``"RectangularPlot"``.
+        format_quantity : str, optional
+            Conversion data function.
+            Available functions are: ``"dB10"``, ``"dB20"``, ``"abs"``, ``"real"``, ``"imag"``,
+            ``"norm"``, ``"ang"``, and ``"deg"``.
+        export_image_path : str, optional
+            Full path for the image file. The default is ``None``, in which case an image in not exported.
+        show : bool, optional
+            Whether to show the plot. The default is ``True``.
+            If ``False``, the Matplotlib instance of the plot is shown.
+        is_polar : bool, optional
+            Whether this plot is a polar plot. The default is ``True``.
 
-        theta_scan2 = math.radians(theta_scan2)
-        phi_scan2 = math.radians(phi_scan2)
+        Returns
+        -------
+        :class:`matplotlib.plt`
+           If ``show`` is ``False``, the plot of the curves is shown.
+           If ``show`` is ``True``, a Matplotlib instance of the plot is returned.
```

Comment:
```suggestion
           If ``show=True``, a Matplotlib instance of the plot is returned.
           If ``show=False``, the plot of the curves is shown.
```
Put the default behavior first.